javascript/code-challenges/challenges-03.test.js

```diff
@@ -0,0 +1,313 @@
+'use strict';
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 1 - Review
+
+Write a function called addTwo that takes in an array and adds two to every value using a for loop. Place the new value in a new array. Return the new array.
+------------------------------------------------------------------------------------------------ */
+
+const addTwo = (arr) => {
+  arr = [1, 2, 4];
+  const newarr = [];
+  for (let i = 0; i < arr.length; i++) {
+    let addtwoCallback = arr[i] + 2;
+    newarr.push(addtwoCallback);
+  }
+  return newarr;
+};
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 2
+
+Write a function named typeNum that, given an array as input, uses filter to return an array containing only the numbers.
+
+For example, typeNum([1, 'bob' ,3]) returns [1,3].
+------------------------------------------------------------------------------------------------ */
+
+const typeNum = (arr) => {
+  const arr2 = arr.filter(arr => typeof arr == 'number');
```

Comment:
From a naming convention standpoint this is just a little confusing.
You're filtering through the array called "arr" and you've given the variable name "arr" to each item in "arr". Typically it's not good policy to have two variables of the same name. Maybe name it "arrData" or just "data"
```js
const typeNum = arr => {
  const arr2 = arr.filter(data => typeof data === 'number');
```
Also, the strict equal "===" is better here. Loose equality, depending on the test cases, might potentially trip you up. Obvioiusly it passes our specific test suite, but just for later. By doing loose equality, it's possible javascript could see the string '2' and say that '2' == 2 is true, whereas '2' === 2 is false.